waiter/src/waiter/scheduler.clj

```diff
@@ -638,3 +643,121 @@
           "WAITER_PASSWORD" (service-id->password-fn service-id)
           "WAITER_SERVICE_ID" service-id
           "WAITER_USERNAME" "waiter"}))
+
+;; Support for tracking service instance launching time stats
+
+(defrecord ServiceInstancesLaunchTracker [service-id known-instance-ids scheduling-instance-timer-contexts starting-instance-context-maps])
```

Comment:
Do we need a `defrecord`?  We created other data structures in this code, but opted to not define names for them. Perhaps better would be a comment that describes the data structure that the loop intends to maintain.

Reply:
Removed.